### PR TITLE
feat: implement Request.prototype.setCacheKey

### PR DESCRIPTION
**Pull request by JakeChampion**
This method is used to set the cache key that will be used when attempting to satisfy this request from a cached response.

It is the equivalent of the VCL property `req.hash`.

The rust sdk version of [this method](https://docs.rs/fastly/latest/src/fastly/experimental.rs.html#183-195)
The header name is from xqd-headers
This [thread](https://fastly.slack.com/archives/CP8H97EDQ/p1664298940822429?thread_ts=1664296201.814199&cid=CP8H97EDQ) is where I learnt that this is the correct approach to take

### CODE REVIEW

File: c-dependencies/js-compute-runtime/js-compute-builtins.cpp [hunk 1]
```diff
@@ -1128,6 +1129,34 @@ JSString *method(JSContext *cx, HandleObject obj) {
...
 
+bool set_cache_key(JSContext *cx, HandleObject self, HandleValue cache_key_val) {
+  MOZ_ASSERT(is_instance(self));
+  size_t key_len;
+  // Convert the key argument into a String following https://tc39.es/ecma262/#sec-tostring
+  JS::UniqueChars keyString = encode(cx, cache_key_val, &key_len);
+  if (!keyString) {
+    return false;
+  }
+  std::string key(keyString.get(), key_len);
```
Comment (elliottt) [inline c-dependencies/js-compute-runtime/js-compute-builtins.cpp:1140]: Does `key` need to copy the bytes from `keyString`, or would it be possible to make a `std::string_view` and use that with `hash256_hex_string` instead?

**Review by elliottt — COMMENTED** [review]
This looks good to me, just one question about whether or not we need to copy the key before hashing it.

File: c-dependencies/js-compute-runtime/js-compute-builtins.cpp (hunk 1 above)
Comment (JakeChampion) [reply c-dependencies/js-compute-runtime/js-compute-builtins.cpp]: I've changed to string_view 👍

**Review by elliottt — APPROVED** [review]
Looks great!

**Pull request merged**